runtime/frame_descriptors.c

```diff
@@ -207,6 +265,7 @@ static void register_frametables_from_stw_single(frametable_array *array)
   for (int i = 0; i < array->ntables; i++)
     new_frametables = cons((intnat*)array->table[i], new_frametables);
 
+  clean_frame_descriptors(&current_frame_descrs);
```

Comment:
One place where to do this would be in major_gc.c, right before or after the call to `caml_code_fragment_cleanup_from_stw_single` within `stw_cycle_all_domains`.

Reply:
I guess that we may want to do the cleanup here (in `register_frametables`) *and* there (in the major GC), instead of doing in just one place. If we did it only in the GC, we avoid unbounded delays but then the addition logic in `register_frametables` may still encounter stale entries.

Reply:
I agree, but I face a difficulty : `major_gc.c` did not depend on `frame_descriptors.c` (the latter appears after in the compilation file order). Changing this order is "*possible*" but will mess with the actual structure of the `Makefile` (macros https://github.com/ocaml/ocaml/blob/733c2c58341302949ce5598273f7679222abef85/Makefile#L1094 and https://github.com/ocaml/ocaml/blob/733c2c58341302949ce5598273f7679222abef85/Makefile#L1159)

Reply:
You are the C expert, can we get away with a `extern void clean_frame_descriptors(void);` declaration somewhere in major_gc.c, without changing the linking order?

Reply:
Hi @gasche, I get a simple workaround that will not deeply modify the `Makefile` but I do not know if it fits the coding standard of the project.
The idea is to simply (conditionally) import the `frame_descriptors.c` file into `major_gc.c` file.
Do you think it is ok ?

Reply:
This sounds like unwanted complexity to me. I propose to just drop my suggestion and keep things as you wrote them initially, where the removal is only forced when adding more frametables later.

Reply:
Ok, so should I drop the commit [6bc2386](https://github.com/ocaml/ocaml/pull/12789/commits/6bc238603cea9a756fa7791492e90f5cf3e08009) ?

Reply:
I guess so. Sorry for the trouble.